espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/Handlers/Finder.java

```diff
@@ -31,95 +21,54 @@
 import static org.hamcrest.Matchers.endsWith;
 
-public class Finder implements RequestHandler {
-
-    public BaseResponse handle(NanoHTTPD.IHTTPSession session, Map<String, String> uriParams)  {
-
-        // If the SessionID is invalid, return InvalidSessionResponse
-        // TODO: Fix SessionID handling redundancies.
-        if (!uriParams.get("sessionId").equals(Session.getGlobalSessionId())) {
-            return new InvalidSessionResponse(uriParams.get("sessionId"));
-        }
-
-        // NanoHTTP requires call to parse body before we can get the parameters
-        // TODO: Move parameter parsing into Router.java
-        try {
-            session.parseBody(new HashMap<String, String>());
-        } catch (NanoHTTPD.ResponseException e) {
-            return new BadRequestResponse("Could not parse parameters");
-        } catch (IOException e) {
-            return new InternalErrorResponse("Internal server error has occurred");
-        }
-
-        AppiumResponse response = new AppiumResponse();
-        Map<String, List<String>> parameters = session.getParameters();
-
-        final Strategy strategy;
-        try {
-            strategy = Strategy.fromString(parameters.get("using").get(0));
-        } catch (final InvalidStrategyException e) {
-            response.setResponse(new Appium());
-            return response;
-        }
-
-        // Get the description
-        String selector = parameters.get("value").get(0);
+public class Finder implements RequestHandler<Locator> {
 
+    public Element handle(Locator locator) throws AppiumException {
         try {
+            if (locator.getUsing() == null) {
+                throw new InvalidStrategyException("Locator strategy cannot be empty");
+            } else if (locator.getValue() == null) {
+                throw new BadParametersException("No locator provided");
+            }
             // Test the selector
-            ViewInteraction matcher = findBy(strategy, selector);
+            ViewInteraction matcher = findBy(locator.getUsing(), locator.getValue());
             matcher.check(matches(isDisplayed()));
 
             // If we have a match, return success
-            Element element = new Element(matcher);
-            response.setValue(element);
-            response.setSessionId(uriParams.get("sessionId"));
-            return response;
-        } catch (InvalidStrategyException e) {
-            return new BadRequestResponse(e.getMessage());
-        } catch (ServerErrorException e) {
-            return new InternalErrorResponse(e.getMessage());
-        } catch (ElementNotFoundException e) {
-            return new BadRequestResponse("Could not find element with " + strategy.getStrategyName() + ": " + selector);
+            return new Element(matcher);
+        } catch (NoMatchingViewException e) {
+            throw new NoSuchElementException("Could not find element with strategy " + locator.getUsing() + " and selector " + locator.getValue());
```

Comment:
yeah, let's make an issue to fix those and add a lint rule that objects to concatenation if possible